src/base/utils/fs.h

```diff
@@ -33,12 +33,40 @@
  * Utility functions related to file system.
  */
 
+#include <iterator>
+#include <memory>
+
 #include <QString>
 
+class QByteArray;
+class QIODevice;
+
 namespace Utils
 {
     namespace Fs
     {
+        // A wrapper class that satisfy LegacyOutputIterator requirement
+        class FileOutputIterator
+            : public std::iterator<std::output_iterator_tag, void, void, void, void>
+        {
+        public:
+            explicit FileOutputIterator(QIODevice &device, const int bufferSize = (4 * 1024));
+            FileOutputIterator(const FileOutputIterator &other) = default;
+            ~FileOutputIterator();
+
+            // mimic std::ostream_iterator behavior
+            FileOutputIterator &operator=(char c);
+            // TODO: make these `constexpr` in C++17
+            FileOutputIterator &operator*();
+            FileOutputIterator &operator++();
+            FileOutputIterator &operator++(int);
+
+        private:
+            QIODevice *m_device;
```

Comment:
You can store it as a reference instead.

Reply:
No, if a reference is used then the class is not copy-constructible.

Reply:
Sorry, cannot understand why...

Reply:
```c++
FileOutputIterator(const FileOutputIterator &other)
{
    // If m_device is a reference, then the following won't work
    m_device = other.m_device;
    // ...
}
```

Reply:
Of course... but the following still should work:
```c++
FileOutputIterator(const FileOutputIterator &other)
    : m_device {other.m_device}
{
}
```

Reply:
Or not?...
Is `other.m_device` of type `const QFileDevice` here?

Reply:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xutility(1269): error C2280: 'Utils::Fs::FileDeviceOutputIterator &Utils::Fs::FileDeviceOutputIterator::operator =(const Utils::Fs::FileDeviceOutputIterator &)': attempting to reference a deleted function
C:\qBittorrent\src\base/utils/fs.h(68): note: compiler has generated 'Utils::Fs::FileDeviceOutputIterator::operator =' here
C:\qBittorrent\src\base/utils/fs.h(68): note: 'Utils::Fs::FileDeviceOutputIterator &Utils::Fs::FileDeviceOutputIterator::operator =(const Utils::Fs::FileDeviceOutputIterator &)': function was implicitly deleted because 'Utils::Fs::FileDeviceOutputIterator' has a data member 'Utils::Fs::FileDeviceOutputIterator::m_device' of reference type
```

Reply:
Ok.